File: frontend/lib/models/service_order.dart
class ServiceOrder {
  final int id;
  final String osNumber;
  final String customerName;
  final String? customerPhone;
  final String? customerEmail;
  final String vehiclePlate;
  final String? vehicleBrand;
  final String? vehicleModel;
  final String? vehicleYear;
  final String description;
  final String? diagnostics;
  final double? estimatedCost;
  final double? finalCost;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final DateTime? finishedAt;

  ServiceOrder({
    required this.id,
    required this.osNumber,
    required this.customerName,
    this.customerPhone,
    this.customerEmail,
    required this.vehiclePlate,
    this.vehicleBrand,
    this.vehicleModel,
    this.vehicleYear,
    required this.description,
    this.diagnostics,
    this.estimatedCost,
    this.finalCost,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.finishedAt,
  });

  factory ServiceOrder.fromJson(Map<String, dynamic> json) {
    return ServiceOrder(
      id: json['id'],
      osNumber: json['osNumber'],
      customerName: json['customerName'],
      customerPhone: json['customerPhone'],
      customerEmail: json['customerEmail'],
      vehiclePlate: json['vehiclePlate'],
      vehicleBrand: json['vehicleBrand'],
      vehicleModel: json['vehicleModel'],
      vehicleYear: json['vehicleYear'],
      description: json['description'],
      diagnostics: json['diagnostics'],
      estimatedCost: json['estimatedCost']?.toDouble(),
      finalCost: json['finalCost']?.toDouble(),
      status: json['status'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      finishedAt: json['finishedAt'] != null
          ? DateTime.parse(json['finishedAt'])
          : null,
    );
  }

  String get statusLabel {
    switch (status) {
      case 'ABERTA':
        return 'Aberta';
      case 'EM_ANALISE':
        return 'Em Análise';
      case 'EM_ANDAMENTO':
        return 'Em Andamento';
      case 'AGUARDANDO_PECAS':
        return 'Aguardando Peças';
      case 'CONCLUIDA':
        return 'Concluída';
      case 'CANCELADA':
        return 'Cancelada';
      case 'ENTREGUE':
        return 'Entregue';
      default:
        return status;
    }
  }
}

File: backend/src/main/java/com/osmech/os/entity/ServiceOrder.java
package com.osmech.os.entity;

import com.osmech.user.entity.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * Entidade que representa uma Ordem de Serviço
 */
@Entity
@Table(name = "service_orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ServiceOrder {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, unique = true, length = 20)
    private String osNumber; // Ex: OS-2026-0001
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @Column(nullable = false, length = 100)
    private String customerName;
    
    @Column(length = 20)
    private String customerPhone;
    
    @Column(length = 100)
    private String customerEmail;
    
    @Column(nullable = false, length = 50)
    private String vehiclePlate;
    
    @Column(length = 50)
    private String vehicleBrand;
    
    @Column(length = 50)
    private String vehicleModel;
    
    @Column(length = 10)
    private String vehicleYear;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(columnDefinition = "TEXT")
    private String diagnostics;
    
    @Column(precision = 10, scale = 2)
    private BigDecimal estimatedCost;
    
    @Column(precision = 10, scale = 2)
    private BigDecimal finalCost;
    
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private OrderStatus status = OrderStatus.ABERTA;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @Column(name = "finished_at")
    private LocalDateTime finishedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
    
    public enum OrderStatus {
        ABERTA,         // Ordem criada
        EM_ANALISE,     // Mecânico analisando
        EM_ANDAMENTO,   // Serviço sendo executado
        AGUARDANDO_PECAS, // Aguardando peças
        CONCLUIDA,      // Serviço finalizado
        CANCELADA,      // Ordem cancelada
        ENTREGUE        // Veículo entregue ao cliente
    }
}
